libutils/ip_address.c

```diff
@@ -1331,3 +1332,22 @@ bool IPAddressIsIPAddress(Buffer *source, IPAddress **address)
     }
     return true;
 }
+
+bool StringIsLocalHostIP(const char *str)
+{
+    const char *regex =
+    // ipv6
+    "(0{0,2}(" // first byte
+        "((:0{1,2}){6})|"
+        "((:0{1,2}){0,5}:)|"
+        "((:0{1,2}){0,4}:(:0{1,2}))|"
+        "((:0{1,2}){0,3}:(:0{1,2}){2})|"
+        "((:0{1,2}){0,2}:(:0{1,2}){3})|"
+        "((:0{1,2}){0,1}:(:0{1,2}){4})|"
+        "(:(:0{1,2}){5})"
+    "):0?1)|" // last byte
+    // ipv4
+    "(127\\.0{1,3}\\.0{1,3}\\.0{0,2}1)";
```

Comment:
I've never seen an IPV4 address written as `127.000.000.001`, not sure if that's even valid. But what I know for sure is that all `127.0.0.0/8` addresses are localhost addresses.